tests/bsim/sh_common.source

```diff
@@ -40,7 +40,7 @@ function check_program_exists() {
 }
 
 function Execute() {
-  EXECUTE_TIMEOUT="${EXECUTE_TIMEOUT:-30}"
+  EXECUTE_TIMEOUT="${EXECUTE_TIMEOUT:-300}"
```

Comment:
Didn't you mention that we very rarely hit this in CI, as an actual issue? 

When running local, the timeout is less of an issue, as a user (I assume) will typically abort it manually if it gets stuck within seconds. 

The problem that this PR fixes is that the majority of the timeouts we see are false positives. The current watchdog values are more of a hinder than a help.

Reply:
what's to say we won't run multiple CI jobs on the same physical machine in the future? In that case they might be 10x slower.

@alwa-nordic you had something for timeout on actual CPU used time, what was it again? We should probably switch to that instead.

Reply:
> Didn't you mention that we very rarely hit this in CI, as an actual issue?
> When running local, 

We hit it as a real problem some times, and some of us use it locally (I do) to run many tests in batch. And sometimes I'm met with stuck tests for one reason or another.

Reply:
> @alwa-nordic you had something for timeout on actual CPU used time, what was it again? We should probably switch to that instead.

`ulimit -t <cpu_time_in_seconds>` is a simple limit on CPU time. The limit is not shared between processes. Each new fork is given the full amount, AFAIK. This does not adjust for the speed of the CPU.

And it does not count IO-blocked time. This part is important, because the real-world-clock timeout scheme will protect against the bsim phy forever waiting for a device that never shows up. Counting CPU-time will not.

I think we should have a watchdog that kills the whole job if the simulation-time is not progressing.

Reply:
> what's to say we won't run multiple CI jobs on the same physical machine in the future?

Running several jobs or VMs in a single physical machine is very normal, but oversubcribing CI runners beyond their number of cores would not be normal or wise. For this kind of jobs what I'd expect the admin would do is split a machine with say 32 cores, into 4 or 8 VMs with 8 or 4 cores each. In that case, more load in the other VMs will slow down a bit your VM booked cores (because memory BW and TDP are shared), but your performance will not plummet to 1/10.

Reply:
@jori-nordic I'm just asking that instead of increasing it to 10x you'd increase it to 2 or 3x